nncf/pruning/model_analysis.py

```diff
@@ -0,0 +1,243 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import networkx as nx
+
+from nncf.nncf_network import NNCFNetwork
+from nncf.pruning.export_helpers import PRUNING_OPERATOR_METATYPES, Convolution, \
+    Concat, StopMaskForwardOps
+from nncf.pruning.utils import is_depthwise_conv, find_next_nodes_not_types
+
+
+# pylint: disable=protected-access
+class NodesCluster:
+    def __init__(self, cluster_id, nodes, nodes_orders):
```

Comment:
Method type hints are missing throughout this file, please add these accordingly.